misc/LADYBUG_USER_STORIES.md

```diff
@@ -3,51 +3,152 @@ Ladybug User Stories
+
+Data in Ladybug reports can be confidential. There are userstories to limit access to Ladybug's functions based on the user's role.
+
+This document focuses on user stories for Ladybug as a part of the Frank!Framework. The following user roles are considered:
+
+**Frank developer:** Someone who writes Frank application. He wants to configure Ladybug as part of his work.
+
+**Frank tester:** Someone who tests or debugs Frank configurations using Ladybug.
+
+**System administrator:** Someone who deploys the Java application and has control over the device and the application server on which the Java application is hosted.
+
+This page presents a few main user stories and groups the other userstories as sub-stories. Each main user story is a separate section.
+
+* [I want to find the report I am interested in](#i-want-to-find-the-report-i-am-interested-in)
+* [I want to understand how the message captured by a report was processed](#i-want-to-understand-how-the-message-captured-by-a-report-was-processed)
+* [I want to turn a report into a test case](#i-want-to-turn-a-report-into-a-test-case)
+* [I want to re-run (test) reports to test my Frank application](#i-want-to-re-run-test-reports-to-test-my-frank-application)
+* [I want to configure whether my Frank application does produce reports](#i-want-to-configure-whether-my-frank-application-does-produce-reports)
+* [I do not want unauthorized access to reports](#i-do-not-want-unauthorized-access-to-reports)
+* [Miscelaneous](#miscelaneous)
+
+# I want to find the report I am interested in
+
+**10:** As a Frank tester, I want to see a table of reports when I open the Ladybug GUI. Each row of the table should be a report and each column should be a metadata attribute. In this table I can search the report of the adapter run that I want to examine.
+
+**20:** As a Frank tester, when I click a row in the table I want to see the corresponding report in the tree view. This allows me to examine it in more detail.
+
+**100:** As a Frank tester, I want to see at least the following metadata attributes in the report table:
+
+* **Name:** The name of the report, which is the name of the outermost start checkpoint.
+* **Correlation id:** See vision.
+* **Storage id:** The id assigned to the report when it was stored.
+* **End time:** The timestamp at which the outermost end checkpoint was created.
+* **Duration:** The time interval between the outermost start checkpoint and the outermost end checkpoint.
+* **Status:** The status with which the pipeline ended, should be SUCCESS or ERROR.
+* **Number of checkpoints**
+* **Estimated memory usage:** TODO: What does this mean?
+* **Storage size:** TODO: What does this mean?
+
+**120:** As a Frank developer, I want to be able to add columns to the report table in addition to the ones shown in story **100**. I am willing to write Spring XML files to achieve this.
+
+TODO: Do I have to specify what metadata extractors we have?
```

Comment:
It would be good to know if we can display columns based on incoming and or outgoing message.

Reply:
Ik heb dit beter opgeschreven zodat hopelijk duidelijk is dat dit kan. Er staat nu ook waarom we dat willen.